### PR TITLE
Add API call to fetch recurring transactions and merchant data

### PR DESCRIPTION
**Pull request by hammem**
Adds a function to fetch the upcoming recurring transactions, as well as the assocaited merchant and account information.

Also did some refactoring to encapsulate common date manipulations using in this class.

### CODE REVIEW

File: monarchmoney/monarchmoney.py [hunk 1]
```diff
@@ -2395,6 +2377,95 @@ async def upload_account_balance_history(
             if resp.status != 200:
                 raise RequestFailedException(f"HTTP Code {resp.status}: {resp.reason}")
 
+    async def get_recurring_transactions(
+        self, start_date: str, end_date: str
```
Comment (grablair) [inline monarchmoney/monarchmoney.py:2381]: I think these start / end dates need to be `Optional[str] = None`, similar to what's done in other methods. I am receiving an error with the no-args call:

```
In [6]: result = asyncio.run(mm.get_recurring_transactions())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 1
----> 1 result = asyncio.run(mm.get_recurring_transactions())

TypeError: MonarchMoney.get_recurring_transactions() missing 2 required positional arguments: 'start_date' and 'end_date'
```

File: monarchmoney/monarchmoney.py [hunk 2]
```diff
@@ -2395,6 +2377,95 @@ async def upload_account_balance_history(
...
+                }
+                __typename
+              }
+            }
+        """
+        )
+
+        variables = {"startDate": start_date, "endDate": end_date}
+
+        if start_date is None ^ end_date is None:
```
Comment (grablair) [inline monarchmoney/monarchmoney.py:2434]: This is also producing an error for me... maybe it's my python version or something funky w/ `ipython`.

```
In [7]: result = asyncio.run(mm.get_recurring_transactions("2024-02-01", "2024-02-29"))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 result = asyncio.run(mm.get_recurring_transactions("2024-02-01", "2024-02-29"))

File /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py:190, in run(main, debug)
    186     raise RuntimeError(
    187         "asyncio.run() cannot be called from a running event loop")
    189 with Runner(debug=debug) as runner:
--> 190     return runner.run(main)

File /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py:118, in Runner.run(self, coro, context)
    116 self._interrupt_count = 0
    117 try:
--> 118     return self._loop.run_until_complete(task)
    119 except exceptions.CancelledError:
    120     if self._interrupt_count > 0:

File /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py:653, in BaseEventLoop.run_until_complete(self, future)
    650 if not future.done():
    651     raise RuntimeError('Event loop stopped before Future completed.')
--> 653 return future.result()

File ~/workspaces/github/monarchmoney/monarchmoney/monarchmoney.py:2434, in MonarchMoney.get_recurring_transactions(self, start_date, end_date)
   2387 query = gql(
   2388     """
   2389     query Web_GetUpcomingRecurringTransactionItems($startDate: Date!, $endDate: Date!, $filters: RecurringTransactionFilter) {
   (...)
   2429 """
   2430 )
   2432 variables = {"startDate": start_date, "endDate": end_date}
-> 2434 if start_date is None ^ end_date is None:
   2435     raise Exception(
   2436         "You must specify both a start_date and end_date, not just one of them."
   2437     )
   2438 elif start_date is None and end_date is None:

TypeError: unsupported operand type(s) for ^: 'NoneType' and 'str'
```

In the above example, the following snippet of line 2434 is highlighted: `None ^ end_date`, suggesting it's being interpreted before the left/right conditions.

For me, this error goes away when I surround both conditions in parentheses (i.e. `(start_date is None) ^ (end_date is None)`)

File: monarchmoney/monarchmoney.py [hunk 3]
```diff
@@ -2395,6 +2377,95 @@ async def upload_account_balance_history(
...
+        return start_of_month.strftime("%Y-%m-%d")
+
+    def _get_end_of_current_month(self) -> str:
+        """
+        Returns the date for the last day of the current month as a string formatted as %Y-%m-%d.
+        """
+        now = datetime.now()
+        _, last_day = calendar.monthrange(now.year, now.month)
+        end_of_month = now.replace(day=last_day)
+        return end_of_month.strftime("%Y-%m-%d")
```
Comment (grablair) [inline monarchmoney/monarchmoney.py:2467]: Yay! Thanks for doing this!

**Review by grablair — CHANGES_REQUESTED** [review]

**Comment by grablair** [comment]
minor: can we add the new method to the README in the next commit?

File: monarchmoney/monarchmoney.py (hunk 1 above)
Comment (hammem) [reply monarchmoney/monarchmoney.py:2381]: ah, good point, fixing!

File: monarchmoney/monarchmoney.py (hunk 2 above)
Comment (hammem) [reply monarchmoney/monarchmoney.py:2434]: oof, yep, it's the order of operations comparing `None` to `end_date`.  fixing...

**Comment by hammem** [comment]
@grablair , should be all set now. Thanks for the review!

**Review by grablair — APPROVED** [review]

**Pull request merged**